content/articles/2019-08-gmailr-1-0-0.Rmarkdown

```diff
@@ -0,0 +1,79 @@
+
+  * There are changes in the auth interface that are not backwards compatible.
+  * The built-in application has been removed.
+  * The functions have all been prefixed with `gm_()`, to avoid name conflicts with functions in other packages (including the base package).
+  
+There is also new functionality that make writing emails with non-ASCII characters more robust, and improved documentation. See the [changelog](http://gmailr.r-lib.org/news/index.html#gmailr-1-0-0) for the full details on this release.
+
+## Auth from gargle
+
+gmailr's auth functionality now comes from the [gargle package](https://gargle.r-lib.org), which provides R infrastructure to work with Google APIs, in general. We've just [blogged about gargle's initial release](https://www.tidyverse.org/articles/2019/08/gargle-hello-world/), so check out that post for more details.
+
+We're adopting gargle for auth in several other packages, such as [bigrquery](https://bigrquery.r-dbi.org) (>= v1.2.0), [googledrive](https://googledrive.r-lib.org)  (>= v1.0.0), and [googlesheets4](https://googlesheets4.tidyverse.org) (currently GitHub-only, successor of googlesheets). This makes new token flows available in these packages, such as [Application Default Credentials](https://www.jhanley.com/google-cloud-application-default-credentials/), and makes auth less idiosyncratic.
+
+### Auth changes the typical user will notice
+
+The functions used for authentication have changed. Use `gm_auth_configure()` to configure your application and then `gm_auth()` to actually authenticate your user with the application.
+
+> OAuth2 tokens are now cached at the user level, by default, instead of in `.httr-oauth` in the current project. We will ask if it's OK to create a new folder to hold your OAuth tokens. We recommend that you delete any vestigial `.httr-oauth` files lying around your gmailr projects and re-authorize gmailr, i.e. get a new token, stored in the new way.
+
+The new strategy makes it harder to accidentally push your tokens to the cloud, easier to use multiple Google identities, and easier to share tokens across projects and packages.
+
+Overall, gmailr has gotten more careful about getting your permission to use a cached token. See the gargle vignette [Non-interactive auth](https://gargle.r-lib.org/articles/non-interactive-auth.html) to learn how to prevent attempts to interact with you, especially the section ["Arrange for an oauth token to be re-discovered"](https://gargle.r-lib.org/articles/non-interactive-auth.html#arrange-for-an-oauth-token-to-be-re-discovered).
+
+gmailr also removes the built-in OAuth "app". This was nessesariy to comply with stricter enforncement of the Google terms of service. It was not possible to comply with the TOS and continue to have a default application embedded in gmailr. See the [setup](http://gmailr.r-lib.org/#setup) section of the readme for how to create a new project and authenticate it with only a few steps.
```

Comment:
```suggestion
gmailr also removes the built-in OAuth "app". This was nessesary to comply with stricter enforcement of the Google terms of service. It was not possible to comply with the TOS and continue to have a default application embedded in gmailr. See the [setup](http://gmailr.r-lib.org/#setup) section of the readme for how to create a new project and authenticate it with only a few steps.
```

Reply:
I'm not sure I would characterize this as a TOS issue. I'd probably say that there are special new rules around the Gmail API that make it extremely difficult to get a Gmail-capable app verified and there is little point in shipping an app that is unverified.

Reply:
Ok, I tweaked the wording and added a link to the google documentation on the subject.